### DISCUSSION TITLE
Toggle Terminal - status bar extension

### DISCUSSION THREAD

**Discussion by krish-r** [microsoft/vscode-discussions#478] [Extension Development QnA] [answered]
Hello, 

I have created a tiny [extension](https://github.com/krish-r/vscode-toggle-terminal) to add terminal toggle icon (shortcut) to the status bar.

<details>
<summary>Demo</summary>

![Demo](https://user-images.githubusercontent.com/54745129/210171062-86860548-cb66-403c-b4e4-db6d15d322bf.gif)
</details>

**Current Logic and Limitations**:
I'm currently using a boolean flag to toggle the terminal (by calling `show()` or `hide()` [terminal](https://code.visualstudio.com/api/references/vscode-api#Terminal) methods).

However, this breaks -
- if the focus is switched from the active terminal to another section on the Panel (like PROBLEMS or OUTPUT).
- if the panel is closed using "Close Panel" option.

And, as a workaround, I need to click the toggle twice to reset the flag to show/focus on the active terminal first (and once more if I need to hide it).
<br>

**What I've tried/found till now**:
* Tried using -
  * `vscode.commands.executeCommand("workbench.action.terminal.toggleTerminal");`, but the command does not seem to toggle the terminal once the focus is on the terminal (found an [old issue](https://github.com/microsoft/vscode/issues/141233#issuecomment-1025880711) discussing about the command behavior).
  * `vscode.commands.executeCommand("workbench.action.togglePanel");` - this also suffers from those limitations.
* Based on this [issue](https://github.com/microsoft/vscode/issues/117980), it looks like currently there isn't an API yet to know if the inline terminal has the focus.
<br>

**What I'm looking for**:
Is there a better way / extension API to do this or emulate the `` CTRL + ` `` shortcut (`View: Toggle Terminal` command: `workbench.action.terminal.toggleTerminal`) behavior since it does not suffer from the above limitations. 
<br>

Thanks!

**Comment by isidorn** [comment]
I do not think there is a better way to do this. @tyriar might know.

**Reply by Tyriar** [reply, marked as answer]
Yes, I think you would need https://github.com/microsoft/vscode/issues/117980

**Comment by krish-r** [comment]
ah! okay. Thanks to both of you for time and inputs. I'll close this discussion and watch out for updates on that issue.
